/*
 * This file is part of the VSS-SDK project.
 *
 * This Source Code Form is subject to the terms of the GNU GENERAL PUBLIC LICENSE,
 * v. 3.0. If a copy of the GPL was not distributed with this
 * file, You can obtain one at http://www.gnu.org/licenses/gpl-3.0/.
 */

package vss_state;

message Pose{
    required float x = 1;
    required float y = 2;
    optional float yaw = 3;
}

message Ball_State{
    required Pose pose = 1;
    optional Pose v_pose = 2;
}

message Robot_State{
    required Pose pose = 1;
    optional Pose v_pose = 2;
}

message Global_State{
    repeated Ball_State balls = 1;
    repeated Robot_State robots_yellow = 2;
    repeated Robot_State robots_blue = 3;
}